.travis.yml

```diff
@@ -13,74 +17,111 @@ python:
+    fi
+  - |
+    if ! test -n "$GRUNT_TESTS"; then
+      pip install -r requirements.txt --quiet
+    fi
   # Install geckodriver required for selenium testing
-  - GECKODRIVER_VERSION=v0.17.0
-  - TAR=geckodriver-$GECKODRIVER_VERSION-linux64.tar
-  - wget https://github.com/mozilla/geckodriver/releases/download/$GECKODRIVER_VERSION/$TAR.gz
-  - gunzip $TAR.gz
-  - tar -xvf $TAR
-  - chmod a+x geckodriver
-  - export PATH=$PATH:`pwd`
+  - |
+    if test -n "$DJANGO_SELENIUM_TESTS"; then
+      GECKODRIVER_VERSION=v0.17.0
+      TAR=geckodriver-$GECKODRIVER_VERSION-linux64.tar
+      wget https://github.com/mozilla/geckodriver/releases/download/$GECKODRIVER_VERSION/$TAR.gz
+      gunzip $TAR.gz
+      tar -xvf $TAR
+      chmod a+x geckodriver
+      export PATH=$PATH:`pwd`
+    fi
+
 env:
   global:
     # These env vars are available to every build
     - PYTHONPATH=$PYTHONPATH:../refinery:../refinery/config
     - DJANGO_SETTINGS_MODULE=config.settings.prod
 
 before_script:
-  # Vagrant is not used on travis: This user is just for the database.
-  - createuser --createdb --no-superuser --no-createrole vagrant
-  - createdb -O vagrant refinery
-
-  - sudo mkdir /data # In vagrant, this is created by puppet; In production, this is an EBS mount.
-  - sudo chown $USER /data
+  - |
+    if ! test -n "$GRUNT_TESTS"; then
+      # Vagrant is not used on travis: This user is just for the database.
+      createuser --createdb --no-superuser --no-createrole vagrant
+      createdb -O vagrant refinery
+      sudo mkdir /data
+      sudo chown $USER /data
+      cd refinery
+      # See http://www.stuartellis.eu/articles/erb/#running-erb-from-the-command-line
+      erb config/config.json.erb > config/config.json
+      python manage.py migrate --noinput
+      cd ../
+    fi
+  - |
+    if ! test -n "$DJANGO_UNIT_TESTS"; then
+      cd refinery
+      npm install -g grunt-cli@0.1.13 bower@1.7.7 --progress false --quiet || ( cat npm-debug.log && false )
+      cd ui
+      npm install --progress false --quiet || ( cat npm-debug.log && false )
+      bower install --config.interactive=false --quiet
+      cd ../../
+    fi
 
-  - cd refinery
-# See http://www.stuartellis.eu/articles/erb/#running-erb-from-the-command-line
-  - erb config/config.json.erb > config/config.json
-  - python manage.py migrate --noinput
-  - npm install -g grunt-cli@0.1.13 bower@1.7.7 --progress false --quiet || ( cat npm-debug.log && false )
-  - cd ui
-  - npm install --progress false --quiet || ( cat npm-debug.log && false )
-  - bower install --config.interactive=false --quiet
-  - cd ../
 script:
-  - set -e # Any error will cause travis to exit early and report a failure.
-
-  # TODO: Not the same behavior as pre-commit hook.
-  # https://github.com/refinery-platform/refinery-platform/issues/1851
-  - flake8 --exclude=migrations,ui ..
+  - "set -e" # Any error will cause travis to exit early and report a failure.
+  - |
+    if ! test -n "$GRUNT_TESTS"; then
```

Comment:
probably just run this under one django test instead of both? Or add a new element to the build matrix?

Reply:
@mccalluc Are you saying just run flake8 under one Django build?